src/objective/regression_objective.hpp

```diff
@@ -695,7 +695,7 @@ class RegressionGammaLoss : public RegressionPoissonLoss {
     } else {
       #pragma omp parallel for schedule(static)
       for (data_size_t i = 0; i < num_data_; ++i) {
-        gradients[i] = static_cast<score_t>(1.0 - label_[i] / std::exp(score[i]) * weights_[i]);
+        gradients[i] = static_cast<score_t>((1.0 - label_[i] / std::exp(score[i])) * weights_[i]);
```

Comment:
```suggestion
        gradients[i] = static_cast<score_t>((1.0 - label_[i] * std::exp(-score[i])) * weights_[i]);
```
Could be a tiny little bit faster.

Reply:
@lorentzenchr 
Regarding timing: the same pattern (i.e. y/exp(p) instead of y*exp(-p) appears multiple time in the same file. My suggestion would be to open a new issue/PR that would change this consistently, along with 1-2 experiments on timings.

Regarding unit tests: I added some in a new R script. On the current master, it fails. With the gamma weight fix it passes.

Reply:
the `exp` dominates timing by far.

Reply:
I agree that we should make the formula in the same file consistent. If we want to change division into multiplication, we should modify all these
https://github.com/microsoft/LightGBM/blob/adf36d766afb6261276d0624f4f5b216d51880c8/src/objective/regression_objective.hpp#L689-L700

Reply:
Done in #4289.